src/runtime/input-manager/input_manager.hpp

```diff
@@ -0,0 +1,162 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "common/events_v2/native_event.hpp"
+#include "common/events_v2/event_target.hpp"
+#include "../constellation.hpp"
+
+// Forward declarations
+class TrContentRuntime;
+
+namespace input_manager
+{
+  /**
+   * Keyboard event data structure for IPC communication
+   */
+  struct TrKeyboardEventData
+  {
+    std::string type;   // "keydown", "keyup", "keypress"
+    std::string key;    // Key value (e.g., "a", "Enter", "Escape")
+    std::string code;   // Physical key code (e.g., "KeyA", "Enter", "Escape")
+    int location;       // Key location (standard, left, right, numpad)
+    bool repeat;        // Whether key is being held down
+    bool altKey;        // Alt modifier state
+    bool ctrlKey;       // Ctrl modifier state
+    bool metaKey;       // Meta modifier state
+    bool shiftKey;      // Shift modifier state
+    int keyCode;        // Legacy keyCode value
+    int charCode;       // Legacy charCode value
+    uint32_t timestamp; // Event timestamp
+  };
+
+  /**
+   * InputManager handles input events from host applications and forwards them
+   * to the appropriate JSAR content instances via IPC.
+   * 
+   * This class provides an abstraction layer between platform-specific input
+   * capture (Unity, Unreal, native, etc.) and JSAR Runtime's event system.
+   */
+  class TrInputManager
+  {
+  public:
+    /**
+     * Construct a new Input Manager instance.
+     * 
+     * @param constellation The constellation instance for IPC communication
+     */
+    explicit TrInputManager(std::shared_ptr<TrConstellation> constellation);
+    virtual ~TrInputManager() = default;
+
+  public:
+    /**
+     * Inject a keyboard event from the host application.
+     * This method converts the host keyboard event into JSAR's internal format
+     * and forwards it to the appropriate content runtime.
+     * 
+     * @param contentId The content runtime ID to send the event to
+     * @param eventData The keyboard event data
+     * @return true if the event was successfully injected, false otherwise
+     */
+    bool injectKeyboardEvent(uint32_t contentId, const TrKeyboardEventData &eventData);
```

Comment:
```suggestion
    bool dispatchKeyboardEvent(const TrKeyboardEventData &eventData);
```
Keyboard event is a global event, no need the content id to use.